src/renderer/Renderer.js

```diff
@@ -221,10 +220,25 @@ export default class Renderer {
     }
 
     renderLayer (renderLayer, drawMetadata) {
+        this.drawMetadata = drawMetadata;
         const dataframes = renderLayer.getActiveDataframes();
         const viz = renderLayer.viz;
         const gl = this.gl;
-        const aspect = this._getAspect();
+
+        dataframes.forEach(dataframe => {
```

Comment:
We can extract this to a method like `this._updateMatrixFor(dataframes)` and avoid making the current method grow more in lines

Reply:
:+1: